recipe/meta.yaml

```diff
@@ -36,7 +36,7 @@ requirements:
     - python-build
     - pip
     - cython >=3.0.4,<3.1
-    - numpy >=2.0.0rc1,<2.3
+    - numpy
```

Comment:
This was causing PyPy to pick up NumPy 2, which it is incompatible with

Think the migrator is doing the heavy lifting for us anyways

So dropped it to see if this would fix the remaining issues

Though happy to discuss if something else needs to be done here